--- net-tools-1.60/lib/pathnames.h.orig	2006-04-12 22:14:07.943885000 -0400
+++ net-tools-1.60/lib/pathnames.h	2006-04-12 22:19:34.072430000 -0400
@@ -42,8 +42,8 @@
 #define _PATH_PROCNET_RARP		"/proc/net/rarp"
 #define _PATH_ETHERS			"/etc/ethers"
 #define _PATH_PROCNET_ROSE_ROUTE	"/proc/net/rose_routes"
-#define _PATH_PROCNET_X25              "/proc/net/x25"
-#define _PATH_PROCNET_X25_ROUTE                "/proc/net/x25_routes"
+#define _PATH_PROCNET_X25		"/proc/net/x25"
+#define _PATH_PROCNET_X25_ROUTE		"/proc/net/x25/route"
 #define _PATH_PROCNET_DEV_MCAST		"/proc/net/dev_mcast"
 
 /* pathname for the netlink device */
