--- net-tools-1.60/ifconfig.c.clear-flag	2008-07-08 11:21:33.000000000 +0200
+++ net-tools-1.60/ifconfig.c	2008-07-08 11:22:55.000000000 +0200
@@ -465,7 +465,7 @@ int main(int argc, char **argv)
 	}
 	if (!strcmp(*spp, "-allmulti")) {
 	    goterr |= clr_flag(ifr.ifr_name, IFF_ALLMULTI);
-	    if (test_flag(ifr.ifr_name, IFF_MULTICAST) > 0)
+	    if (test_flag(ifr.ifr_name, IFF_ALLMULTI) > 0)
 	    	fprintf(stderr, _("Warning: Interface %s still in ALLMULTI mode.\n"), ifr.ifr_name);
 	    spp++;
 	    continue;
@@ -488,7 +488,7 @@ int main(int argc, char **argv)
 	}
 	if (!strcmp(*spp, "-dynamic")) {
 	    goterr |= clr_flag(ifr.ifr_name, IFF_DYNAMIC);
-	    if (test_flag(ifr.ifr_name, IFF_MULTICAST) > 0)
+	    if (test_flag(ifr.ifr_name, IFF_DYNAMIC) > 0)
 	    	fprintf(stderr, _("Warning: Interface %s still in DYNAMIC mode.\n"), ifr.ifr_name);
 	    spp++;
 	    continue;
@@ -547,7 +547,7 @@ int main(int argc, char **argv)
 
 	if (!strcmp(*spp, "-broadcast")) {
 	    goterr |= clr_flag(ifr.ifr_name, IFF_BROADCAST);
-	    if (test_flag(ifr.ifr_name, IFF_MULTICAST) > 0)
+	    if (test_flag(ifr.ifr_name, IFF_BROADCAST) > 0)
 	    	fprintf(stderr, _("Warning: Interface %s still in BROADCAST mode.\n"), ifr.ifr_name);
 	    spp++;
 	    continue;
@@ -676,7 +676,7 @@ int main(int argc, char **argv)
 	}
 	if (!strcmp(*spp, "-pointopoint")) {
 	    goterr |= clr_flag(ifr.ifr_name, IFF_POINTOPOINT);
-	    if (test_flag(ifr.ifr_name, IFF_MULTICAST) > 0)
+	    if (test_flag(ifr.ifr_name, IFF_POINTOPOINT) > 0)
 	    	fprintf(stderr, _("Warning: Interface %s still in POINTOPOINT mode.\n"), ifr.ifr_name);
 	    spp++;
 	    continue;
